packages/server/src/server.ts

```diff
@@ -153,12 +153,6 @@ export class Server extends EventEmitter {
       return this.write(res, status_code, body)
     }
 
-    try {
-      await this.options.onIncomingRequest?.(req, res)
-    } catch (err) {
-      return onError(err)
-    }
-
```

Comment:
What if we only add `id` to this callback? This way we don't need to edit every handler.
```
    try {
      const id = this.handlers.PATCH.getFileIdFromRequest(req)
      await this.options.onIncomingRequest?.(req, res, id !== false ? id : undefined)
    } catch (err) {
      return onError(err)
    }
```

Reply:
I can do that - the only thing is that it will be called twice this way

Reply:
Also, this approach wouldn't work for the POST endpoint, which creates the URL